soroban-env-host/src/storage.rs

```diff
@@ -242,6 +242,46 @@ impl Storage {
         self.put_opt(key, Some(val), budget)
     }
 
+    pub fn bump(
+        &mut self,
+        key: &Rc<LedgerKey>,
+        min_ledgers_to_live: u32,
+        ledger_num: u32,
+        budget: &Budget,
+    ) -> Result<(), HostError> {
+        // A expiration ledger bump is considered to be a read only operation. When
+        // concurrent Soroban transactions are added conflicts will be reconciled upstream.
```

Comment:
Oh! Interesting. I think this will need special handling in the non-concurrent case as well: the core-side path that examines the modified ledger entries coming out of the host will (I think?) fault if there's a write outside the write-part of the footprint. I'm tempted to spell this out in more detail in the data structure, eg. change the implementation of the storage object or footprint to track "commutative writes" separately from reads and writes, because that's really what the claim here is..

Reply:
Hmm good point. Core would just end up ignoring those bumps. I added a commit for a new access type. This is a little annoying though because the xdr Footprint will need to be updated to add a third vector for these entries. I'd prefer to do this in a way that wouldn't require an xdr update. Maybe we could include the access type change, but then have enforce_access upgrade AccessType from ReadOnly to CommutativeWrite on read only bumps?

Reply:
I pushed another commit to avoid the xdr commit.

Reply:
I'm not sure what you mean by "avoid the XDR commit" here?

Reply:
I meant I made a change where we would avoid having a third vector in the LedgerFootprint for the CommutativeWrite entries.

Reply:
this got deferred for now, thanks.